### ISSUE TITLE
Default template duplicates namespaces documentation.

### ISSUE THREAD

**Issue by rcosnita** [jsdoc3/jsdoc#1059]
With the latest version of master branch, default template duplicates each generated namespace documentation. Is there something which must be fixed or am I doing something wrong? Below you can find a documented namespace where the problem occurs:

```javascript
    /**
     * This namespace holds all SDK classes used for creating new components.
     *
     * ![BCAPI Components Overview](assets/ui-sdk-bcapi-components.png)
     * 
     * @namespace BCAPI.Components
     */
    BCAPI.Components = {};
```

The generated html contains the namespace documentation generated twice. Same problem occurs for classes.

**Comment by hegemonic** [comment]
I can't reproduce the issue using the sample you provided. Can you provide a complete code sample that demonstrates the issue?

(Also, FYI, you can just use `@namespace` rather than `@namespace BCAPI.Components`. JSDoc should figure out the name of the namespace from the code.)

**Comment by hegemonic** [comment]
Closing due to lack of information.

**Comment by rcosnita** [comment]
I tried the latest jsdoc3 version on a very simple project (a single namespace containing a simple class and a webcomponent) and indeed it works correctly. I will further investigate the issue and follow up with more details once I figure out why I have the issue in my project.

Thanks for your help.

**Comment by rcosnita** [comment]
Ok, I finally managed to figure out the root cause of the problem: https://github.com/rcosnita/jsdoc3-generate-duplicates (provides a full example with generated doc files included)

In jsdoc.conf, I configured **source** as:

```json
{
...
     "source": {
            "includePattern": ".*\\.(js|html)$",
            "include": [
                "./src",
                "./src/test"
            ]
     }
...
}
```

Every namespace / class which is defined in ./src/test/*.js has documentation duplicated. The same effect is obtained when you accidentally define the same namespace / class twice.

```javascript
(function() {
      /**
       * Test.
       * @namespace Sample.NS
       */
     var namespace = {};

      /**
       * Test.
       * @namespace Sample.NS
       */
     var namespace2 = {};
})();
```

I think a good solution would to raise an error when duplicate definitions are found. Moreover, developer should be informed about the files which generated the issue.

Another improvement would be to remove only top folders defined in **source** entry of jsdoc.conf:

e.g ./src and ./src/test should be kept as ./src

I can fix both bugs if you think it's important.
